lib/Sema/ConstraintSystem.cpp

```diff
@@ -5675,6 +5699,50 @@ Type ConstraintSystem::getVarType(const VarDecl *var) {
   });
 }
 
+void ConstraintSystem::setContextualType(
+    ASTNode node, TypeLoc T, ContextualTypePurpose purpose,
+    FreeTypeVariableBinding allowFreeTypeVariables) {
```

Comment:
I think we need a special version of this for `SolutionApplicationTarget &` instead of `ASTNode` because _if_ conversion type gets modified `SolutionApplicationTarget` needs to know about that via `setExprConversionType{Loc}` otherwise solution application would crash because of a stale type.